testkit/scaladsl/src/main/scala/com/lightbend/lagom/scaladsl/testkit/ServiceTest.scala

```diff
@@ -122,11 +140,31 @@ object ServiceTest {
     @ApiMayChange
     def withSsl(): Setup = withSsl(true)
 
+    /**
+     * Sets the HTTP port the server should run on.
+     *
+     * By default, a dynamically assigned port will be used.
+     *
+     * @param port The port to assign the server to.
+     * @return A copy of this setup.
+     */
+    def withPort(port: Int): Setup
+
+    /**
+     * The server HTTP port.
+     */
+    def port: Int
+
     /**
      * Whether Cassandra is enabled.
      */
     def cassandra: Boolean
 
+    /**
+     * The Cassandra server port.
+     */
+    def cassandraPort: Int
```

Comment:
I'm ok to add this, but another solution is to run Cassandra on a docker (eventually with Testcontainers) and have full control of it. 

In a hindsight, I find it was a mistake to have servers started in test (except for the lagom server itself). Users struggle with that more often than not. Naive port assignment strategies have caused "Address already in use" errors in our build. We are opening the door for the same to happen in user land code.

In addition to that, when JDBC support was added the feature was not symmetric because we can't start SQL database from inside the ServiceTest.

Reply:
I agree with that, in fact we use Testcontainers ourselves instead of relying on anything in TestKit (other than the Lagom server). I mostly added this for #1584 -- I actually don't have this use-case myself. If the plan is to eventually remove this, maybe I should not add these now.

Reply:
We don't have concrete plans to remove it, although I won't be against. 

If you don't have such a use-case, let's then not add it. The less API surface the better the day we start to deprecate it and recommend docker instead.